metrics/deploy/prometheus-ocs-rules.yaml

```diff
@@ -181,12 +181,14 @@ spec:
     rules:
     - alert: ClusterObjectStoreState
       annotations:
-        description: RGW endpoint of the Ceph object store is in a failure state for more than 15s. Please check the health of the Ceph cluster.
-        message: Cluster Object Store is in unhealthy state. Please check Ceph cluster health.
+        description: RGW endpoint of the Ceph object store is in a failure state or one or more Rook Ceph RGW deployments have fewer ready replicas than required for more than 15s. Please check the health of the Ceph cluster and the deployments.
+        message: Cluster Object Store is in unhealthy state or number of ready replicas for Rook Ceph RGW deployments is less than the desired replicas.
         severity_level: error
         storage_type: RGW
       expr: |
         ocs_rgw_health_status{job="ocs-metrics-exporter"} == 2
+        or
+        kube_deployment_status_replicas_ready{deployment=~"rook-ceph-rgw-.*"} < kube_deployment_spec_replicas{deployment=~"rook-ceph-rgw-.*"}
```

Comment:
Does this function check for Pod status or deployment status??

Reply:
it checks for a ReplicaSet, number of replica's ready vs number of replicas in the deployment spec.

Reply:
can u please confirm when replica set is ready means Pod Status is Ready

Reply:
> can u please confirm when replica set is ready means Pod Status is Ready

reaplicaset see's the number of pods ready, so if the pod is not ready it'll not be counted for `kube_deployment_status_ready`

Reply:
Thanks for confirmation